server/src/main/java/org/opensearch/tasks/TaskManager.java

```diff
@@ -202,6 +223,9 @@ public void cancel(CancellableTask task, String reason, Runnable listener) {
      */
     public Task unregister(Task task) {
         logger.trace("unregister task for id: {}", task.getId());
+        if (taskStatsConsumersEnabled == true && task.getResourceStats().isEmpty() == false) {
```

Comment:
[NIT] ```if (taskStatsConsumersEnabled && !task.getResourceStats().isEmpty())``` {}

Reply:
Modified to ```if (taskStatsConsumersEnabled && task.getResourceStats().isEmpty() == false)``` as OpenSearch uses the pattern ```== false```over ```!```